modules/articles/client/controllers/articles.client.controller.js

```diff
@@ -1,68 +1,52 @@
-'use strict';
+/**
+ * Articles Controller
+ */
 
-// Articles controller
-angular.module('articles').controller('ArticlesController', ['$scope', '$stateParams', '$location', 'Authentication', 'Articles',
-	function($scope, $stateParams, $location, Authentication, Articles) {
-		$scope.authentication = Authentication;
+(function () {
+	'use strict';
 
-		// Create new Article
-		$scope.create = function() {
-			// Create new Article object
-			var article = new Articles({
-				title: this.title,
-				content: this.content
-			});
+	angular
+		.module('articles')
+		.controller('ArticlesController', ArticlesController);
 
-			// Redirect after save
-			article.$save(function(response) {
-				$location.path('articles/' + response._id);
+	ArticlesController.$inject = ['$state', 'Authentication', 'articleResolve'];
 
-				// Clear form fields
-				$scope.title = '';
-				$scope.content = '';
-			}, function(errorResponse) {
-				$scope.error = errorResponse.data.message;
-			});
-		};
+	function ArticlesController($state, Authentication, articleResolve) {
+		var vm = this;
+		vm.authentication = Authentication;
+		vm.article = articleResolve;
+		vm.save = save;
+		vm.remove = remove;
 
-		// Remove existing Article
-		$scope.remove = function(article) {
-			if (article) {
-				article.$remove();
+		// Save Article
+		function save() {
+			var article = vm.article;
 
-				for (var i in $scope.articles) {
-					if ($scope.articles[i] === article) {
-						$scope.articles.splice(i, 1);
-					}
-				}
+			if (article._id)
```

Comment:
@rhutchison Following up from Gitter, this was the line I was referring to. Here we check if the article has an ID to then decide to save or update. I might be misinterpreting the Style guide, but one could see this as being too tied to the data service - the controller shouldn't know what having an _id means. It could be refactored so that the controller's save() calls a method on a service named saveOrCreate() which then examines the object to determine what path to take. I understand this is probably a matter of opinion but wanted to let you know my thoughts.